src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java

```diff
@@ -0,0 +1,82 @@
+import meteordevelopment.meteorclient.utils.Utils;
+import meteordevelopment.meteorclient.utils.render.color.Color;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.regex.Matcher;
+
+public class ProxiesImportScreen extends WindowScreen {
+
+    private final File file;
+    public ProxiesImportScreen(GuiTheme theme, File file) {
+        super(theme, "Import Proxies");
+        this.file = file;
+        this.onClosed(() -> {
+            if (parent instanceof ProxiesScreen screen) {
+                screen.dirty = true;
+            }
+        });
+    }
+
+    @Override
+    public void initWidgets() {
+        if (file.exists() && file.isFile()) {
+            add(theme.label("Importing proxies from " + file.getName() + "...").color(Color.GREEN));
+            WVerticalList list = add(theme.section("Log", false)).widget().add(theme.verticalList()).expandX().widget();
+            Proxies proxies = Proxies.get();
+            try {
+                int pog = 0, bruh = 0;
+                for (String line : Files.readAllLines(file.toPath())) {
+                    Matcher matcher = Proxies.PROXY_PATTERN.matcher(line);
+                    if (matcher.matches()) {
+                        Proxy proxy = new Proxy();
+                        proxy.address = matcher.group(2).replaceAll("\\b0+\\B", "");
+                        // should be safe to parse because of regex matching
+                        proxy.port = Integer.parseInt(matcher.group(3));
+                        proxy.name = matcher.group(1) != null ? matcher.group(1) : proxy.address + ":" + proxy.port;
+                        proxy.type = matcher.group(4) != null ? ProxyType.parse(matcher.group(4)) : ProxyType.Socks4;
+                        if (proxies.add(proxy)) {
+                            list.add(theme.label("Imported proxy: " + proxy.name).color(Color.GREEN));
+                            pog++;
+                        } else {
+                            list.add(theme.label("Proxy already exists: " + proxy.name).color(Color.ORANGE));
+                            bruh++;
+                        }
+                    } else {
+                        list.add(theme.label("Invalid proxy: " + line).color(Color.RED));
+                        bruh++;
+                    }
+                }
+                add(theme
+                    .label("Successfully imported " + pog + "/" + (bruh + pog) + " proxies.")
+                    .color(Utils.lerp(Color.RED, Color.GREEN, (float) pog / (pog + bruh)))
+                );
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            add(theme.label("Invalid File!"));
+        }
+        add(theme.horizontalSeparator()).expandX();
+        WButton btnBack = add(theme.button("Back")).expandX().widget();
+        btnBack.action = () -> MeteorClient.mc.setScreen(parent);
```

Comment:
While this works to close the screen it is better practise to call the `onClose()` method.

Sorry for not pointing it out earlier.

Reply:
`onClose` sounded like the close listener which is why I didn't use it. Also, there are other parts in the code where I found this method of switching screens which is why I used it.

Reply:
Oh